### ISSUE TITLE
Introduce Permission Class Validator: @CheckPermissionIf

### ISSUE THREAD

**Issue by ottovw** [kittgen/kittgen-nestjs#5]
```
class CreateBlogPostDto {

  @CheckPermissionIf(s => s == PublicationState.Published, PublishBlogPostAction)
   publicationState: PublicationState

}
```

**Comment by ottovw** [comment]
@edgarmueller  and i discussed, that this should not be part of class validator but rather of the Guard

**Comment by edgarmueller** [comment]
Addressed via #8
